src/audio/audio.hpp

```diff
@@ -85,14 +85,15 @@ class Audio
 	/**
 	 * Emits the requested response.
 	 *
-	 * @param code The code of the response to emit, if possible.
-	 * @param sink The ResponseSink to which the response shall be sent.
-	 *   May be nullptr, in which case Emit should be a no-operation.
-	 * @param id The ID of the connection to which the ResponseSink should
-	 *   route the response.  May be 0 (the default), for all (broadcast).
+	 * This method does NOT support emitting composite responses.  For
+	 * example, `Emit("/player", x)` shall be ignored.  This functionality
+	 * is provided in Player.
+	 *
+	 * @param path The path of the response to emit, if possible.
+	 * @param broadcast If true, the emission is an update broadcast.
+	 * @return A pointer to the response, if it exists.
 	 */
-	virtual void Emit(Response::Code code, const ResponseSink *sink,
-	                  size_t id = 0);
+	virtual std::unique_ptr<Response> Emit(const std::string &path, bool broadcast);
```

Comment:
how do you mean, not preserved?  ownership should be transferring from Emit to the thing that's calling it and staying there, unless I fucked up